src/main/scala/firrtl/Utils.scala

```diff
@@ -220,25 +220,26 @@ object Utils extends LazyLogging {
       case (t1: SIntType, t2: SIntType) => if (flip1 == flip2) Seq((0, 0)) else Nil
       case (t1: BundleType, t2: BundleType) =>
         def emptyMap = Map[String, (Type, Orientation, Int)]()
-        val t1_fields = ((t1.fields foldLeft (emptyMap, 0)){case ((map, ilen), f1) =>
-          (map + (f1.name -> (f1.tpe, f1.flip, ilen)), ilen + get_size(f1.tpe))})._1
-        ((t2.fields foldLeft (Seq[(Int, Int)](), 0)){case ((points, jlen), f2) =>
+        val t1_fields = (t1.fields foldLeft(emptyMap, 0)) { case ((map, ilen), f1) =>
+          (map + (f1.name ->(f1.tpe, f1.flip, ilen)), ilen + get_size(f1.tpe))
+        }._1
+        (t2.fields foldLeft(Seq[(Int, Int)](), 0)) { case ((points, jlen), f2) =>
           t1_fields get f2.name match {
             case None => (points, jlen + get_size(f2.tpe))
-            case Some((f1_tpe, f1_flip, ilen))=>
+            case Some((f1_tpe, f1_flip, ilen)) =>
               val f1_times = times(flip1, f1_flip)
               val f2_times = times(flip2, f2.flip)
               val ls = get_valid_points(f1_tpe, f2.tpe, f1_times, f2_times)
-              (points ++ (ls map {case (x, y) => (x + ilen, y + jlen)}), jlen + get_size(f2.tpe))
+              (points ++ (ls map { case (x, y) => (x + ilen, y + jlen) }), jlen + get_size(f2.tpe))
           }
-        })._1
+        }._1
       case (t1: VectorType, t2: VectorType) =>
         val size = math.min(t1.size, t2.size)
-        (((0 until size) foldLeft (Seq[(Int, Int)](), 0, 0)){case ((points, ilen, jlen), _) =>
+        ((0 until size) foldLeft(Seq[(Int, Int)](), 0, 0)) { case ((points, ilen, jlen), _) =>
```

Comment:
For future review (since your changes do make it slightly better), this fold is still a bit of a mess.
The map-case should be changed to a for-comprehension.
Instead of binding to ls, can do
```scala
val newPoints = for((x,y) <- get_valid_points(t1.tpe, t2.tpe, flip1, flip2)) yield (x + ilen, y + jlen)
```
Also, a lot of style guides discourage the use of ._1 to access Bundle fields.
The many small appends isn't very efficient for a Seq. Doing a scanLeft then flatten may be better.